openpgp/packet/private_key.go

```diff
@@ -174,27 +229,27 @@ func (pk *PrivateKey) Serialize(w io.Writer) (err error) {
 			'G', 'N', 'U', // "GNU" as a string
 			1, // Extension type 1001 (minus 1000)
 		})
+	} else if pk.Encrypted {
+		_, err = buf.Write([]byte{
+			254,             // SHA-1 Convention
+			byte(pk.cipher), // Encryption scheme
+		})
+		if err != nil {
```

Comment:
Like I said at the beginning, I'm a Golang noob. So, if a method returns `error`, I'm going to check it. And you are correct that the current `Buffer.Write()` only returns `nil` for `error`. Is there any harm in checking?